lib/fastlane/plugin/fueled/actions/is_build_necessary.rb

```diff
@@ -0,0 +1,76 @@
+require_relative '../helper/fueled_helper'
+
+module Fastlane
+  module Actions
+    module SharedValues
+      IS_BUILD_NECESSARY = :IS_BUILD_NECESSARY
+    end
+
+    class IsBuildNecessaryAction < Action
+      def self.run(params)
+        last_config_tag = other_action.last_git_tag(pattern: "v*-#{params[:build_config]}") || ""
```

Comment:
In case we don't have a tag, I think this should be `| | "HEAD"` instead 🤔? 
An empty string will break the command below

Reply:
Or maybe it's better to make this action not depending on our tag format?

Getting the date for the last added tag then use `sh(git rev-list --since=#{last_added_tag_timestamp} --count)` seems like a general solution

Reply:
@hadiidbouk  Might create(rare) an edge case of similar timestamped commits.

Reply:
@notbenoit small optimisation here, it would nice if we can pull out these commands

```
last_config_tag = other_action.last_git_tag(pattern: "v*-#{params[:build_config]}") || "HEAD"
commit_count = sh("git rev-list #{last_config_tag}.. --count").strip
```
 into a function and reverse the predicate in the `if`

`if params[:force_necessary] || commit_count() > 0`

Reply:
I'll look into the refactor. However there's definitely an edge case that needs to be handled when there's no tag. And using `HEAD` as a default value would skip the build as we would compare `HEAD` with `HEAD`

Reply:
>Or maybe it's better to make this action independent of our tag format?

We need to skip builds of similar configuration. I've also been looking at GH `cache` action to see how we could benefit from it, will keep you posted.